File: Medium/Python/1817. Finding the Users Active Minutes.py
class Solution(object):
    def findingUsersActiveMinutes(self, logs, k):
        uam = [0] * k

        # Store each user's activity into a dict
        activity = {}
        for id, time in logs:
            if id not in activity:
                activity[id] = set()
            activity[id].add(time)
        
        # Count the size of each user's activity, and store it
        # in the uam list
        for i in activity:
            uam[len(activity[i]) - 1] += 1
        
        # Return the uam list
        return uam

File: Medium/Java/1817. Finding the Users Active Minutes.java
class Solution {
    public int[] findingUsersActiveMinutes(int[][] logs, int k) {
        int[] uam = new int[k];
        
        // Store each user's activity into a map
        Map<Integer, Set<Integer>> activity = new HashMap<>();
        for(int[] i : logs){
            if(!activity.containsKey(i[0]))
                activity.put(i[0], new HashSet<>());
            activity.get(i[0]).add(i[1]);
        }

        // Count the size of each user's activity, and store it
        // in the uam array
        for(int key : activity.keySet()){
            uam[activity.get(key).size() - 1]++;
        }

        // Return the uam array
        return uam;
    }
}
